File: app/src/main/java/awais/instagrabber/repositories/responses/directmessages/DirectItemLinkContext.java
package awais.instagrabber.repositories.responses.directmessages;

public class DirectItemLinkContext {
    private final String linkUrl;
    private final String linkTitle;
    private final String linkSummary;
    private final String linkImageUrl;

    public DirectItemLinkContext(final String linkUrl,
                                 final String linkTitle,
                                 final String linkSummary,
                                 final String linkImageUrl) {
        this.linkUrl = linkUrl;
        this.linkTitle = linkTitle;
        this.linkSummary = linkSummary;
        this.linkImageUrl = linkImageUrl;
    }

    public String getLinkUrl() {
        return linkUrl;
    }

    public String getLinkTitle() {
        return linkTitle;
    }

    public String getLinkSummary() {
        return linkSummary;
    }

    public String getLinkImageUrl() {
        return linkImageUrl;
    }
}

File: app/src/main/java/awais/instagrabber/repositories/responses/directmessages/DirectItemLinkContext.kt
package awais.instagrabber.repositories.responses.directmessages

import java.io.Serializable

data class DirectItemLinkContext(
    val linkUrl: String? = null,
    val linkTitle: String? = null,
    val linkSummary: String? = null,
    val linkImageUrl: String? = null
) : Serializable